vdp/pipeline/v1beta/pipeline.proto

```diff
@@ -1938,3 +1939,193 @@ message LookUpPipelineAdminResponse {
 }
+
+// ListPipelineRunsRequest is the request message for ListPipelineRuns.
+message ListPipelineRunsRequest {
+  // The namespace to list pipeline runs for.
+  string namespace_id = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // Pipeline ID
+  string pipeline_id = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // The page number to retrieve.
+  int32 page = 3 [(google.api.field_behavior) = OPTIONAL];
+
+  // The maximum number of items per page to return. The default and cap values
+  // are 10 and 100, respectively.
+  int32 page_size = 4 [(google.api.field_behavior) = OPTIONAL];
+
+  // View allows clients to specify the desired run view in the response.
+  // The basic view excludes input / output data.
+  optional Pipeline.View view = 5 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// ListPipelineRunsResponse is the response message for ListPipelineRuns.
+message ListPipelineRunsResponse {
+  // The list of pipeline runs.
+  repeated PipelineRun pipeline_runs = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The total number of pipeline runs matching the request.
+  int64 total_size = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The current page number.
+  int32 page = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The number of items per page.
+  int32 page_size = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+// ListComponentRunsRequest is the request message for ListComponentRuns.
+message ListComponentRunsRequest {
+  // The unique identifier of the pipeline run to list component runs for.
+  string pipeline_run_id = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // The page number to retrieve.
+  optional int32 page = 3 [(google.api.field_behavior) = OPTIONAL];
+
+  // The maximum number of items per page to return. The default and cap values
+  // are 10 and 100, respectively.
+  optional int32 page_size = 4 [(google.api.field_behavior) = OPTIONAL];
+}
+
+// ListComponentRunsResponse is the response message for ListComponentRuns.
+message ListComponentRunsResponse {
+  // The list of component runs.
+  repeated ComponentRun component_runs = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The total number of component runs matching the request.
+  int64 total_size = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The current page number.
+  int32 page = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The number of items per page.
+  int32 page_size = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+
+// FileReference represents metadata for a file.
+message FileReference {
+  // Name of the file.
+  string name = 1 [(google.api.field_behavior) = REQUIRED];
+
+  // Format of the file (e.g., PDF, TXT, JPG).
+  string type = 2 [(google.api.field_behavior) = REQUIRED];
+
+  // Size of the file in bytes.
+  int64 size = 3 [(google.api.field_behavior) = REQUIRED];
+
+  // URL of the file (e.g., S3 URL).
+  string url = 4 [(google.api.field_behavior) = REQUIRED];
```

Comment:
> There's a discrepancy here between model and pipeline. If I understand correctly, in model the backend will fetch the input / output from the file storage and aggregate them as part of the request, while in pipeline the frontend will be responsible of this.

confirmed with Alex that backend will fetch data from minio and return input/output in response

Reply:
> Another question: in the pipeline builder, (I think) the frontend knows the type of the output object, so it can render it as JSON, image, audio, etc. How is this communicated to the frontend in this case?

will confirm with Alex and include schema in response if needed